pkg/routeagent_driver/handlers/ovn/handler.go

```diff
@@ -144,6 +144,12 @@ func (ovn *Handler) RemoteEndpointCreated(endpoint *submV1.Endpoint) error {
 	}
 
 	if ovn.isGateway {
+		for _, subnet := range endpoint.Spec.Subnets {
```

Comment:
We will still have to handle the out of order events which can trigger race conditions. Basically, we have to look at the lastProcessedTime and process/ignore the event similar to how we are doing in kubeproxy handler - https://github.com/submariner-io/submariner/blob/0c3e043a374c6a793c564a9d98fa8fb8990b310e/pkg/routeagent_driver/handlers/kubeproxy/endpoint_handler.go#L108

But anyway, we can use a separate PR for that.

Reply:
sure will do as a follow up pr